internals/daemon/daemon.go

```diff
@@ -213,7 +218,8 @@ func (c *Command) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// not good: https://github.com/canonical/pebble/pull/369
 	var user *UserState
 	if _, isOpen := access.(OpenAccess); !isOpen {
-		user, err = userFromRequest(c.d.state, r, ucred)
+		basicAuthUsername, basicAuthPassword, _ := r.BasicAuth()
```

Comment:
Given the context / scope of these variables, just `username` and `password` would be fine:
```suggestion
		username, password, _ := r.BasicAuth()
```